### DISCUSSION TITLE
[FEATURE REQUEST] With the feature of assigning workspaces to a website and assigning containers to a website, when we want to open the website URL, the website will open in the workspaces we assign and in the container we assign.

### DISCUSSION THREAD

**Discussion by universish** [zen-browser/desktop#2961] [Ideas]
🔢 🥇 💯  The feature I am talking about is that we can assign a Workspaces to a website. When we try to open the website, the website will open in the workspaces we have assigned. It will also open in the container we assigned to the website. If we have not assigned a container to the website, it will open in the container we assigned to the workspaces when we created the workspaces.

If the feature of assigning containers to workspaces is added (I created this feature request. This feature is available in Floorp browser). 

For this feature to work, you first need to install the Firefox Multi-Account Containers plugin. Then you create a container for your workspaces. For example, you should create a git container for git websites (e.g. github gitlab etc.). When creating a new workspaces you assign a container for workspaces and create workspaces. You also change the name of workspaces to git. Then you open github in this workspaces. It automatically runs it in the container named git. This is a very useful feature. 

Note: This is not the feature I requested in this issue. I have already created an issue for this feature request. In this issue, I am requesting another feature. The feature I am talking about here works together with the feature I mentioned in the previous feature request. See also: #2912 

To assign a container to a website, we open the website without a container. Then left click on the “Always Open Site in Container” button created by the Firefox Multi-Account Containers add-on. Then we select the container we want to assign. The website opens in a new tab in that container. After that, every time we open the website, it will always open in that container. 

🔢   **The feature I am talking about is that we can assign a Workspaces to a website. When we try to open the website, the website will open in the workspaces we have assigned. It will also open in the container we assigned to the website. If we have not assigned a container to the website, it will open in the container we assigned to the workspaces when we created the workspaces.** 

This feature is very useful and improves security. It is also a very useful feature to increase security. We can easily prevent cross-site cookies. 

Can you add this feature to zen-browser?

**Comment by kakato10** [comment]
I think that if you can assign a Profile to a Workspace, it would be enough to be able to assign a Workspace for a website to achieve the behavior you described

**Reply by Fxxxlei** [reply]
Maybe you are using the wrong terminology or mixing things up.
Profiles are not containers and you cannot assign profiles to workspaces, if anything it's the other way around.
But you can assign containers to wokspaces, you can set URLs to automatically open in specific containers with an extension (Firefox Multi-Account Containers) and you can configure that tabs should be moved to a workspace with their container assigned to it.

Combining those, I think this should cover the feature.

**Reply by loicfavory** [reply]
> But you can assign containers to wokspaces

@Fxxxlei how can I do that ?

**Reply by loicfavory** [reply]
OK I found it, sorry for the quote

**Comment by HarshPatel5940** [comment]
Does anyone have a work-around or updates for this?

**Comment by FlakyBlueJay** [comment]
I'd suggest changing the title to something a bit more succinct like "automatic workspace and container assignment". 

Containers aside (spaces can automatically set containers for tabs, if not the container extension can do it), and security benefits aside (never did this for security really) I did like the automatic assignment feature when I used extensions like Simple Tab Groups and Sidebery, and on the Chromium side, I think Vivaldi has this too now. Automatic space assignment would be a great feature to have for organisation tbh.

**Reply by HarshPatel5940** [reply]
I agree. Much cleaner.

**Comment by wasilaq** [comment]
I can have certain websites automatically open in certain containers on Firefox. Can't find an easy way to set that up on Zen. Would be really convenient to have this feature

**Reply by HarshPatel5940** [reply]
How do you do in Firefox? Maybe that can make it workable in zen

**Reply by wasilaq** [reply]
looking at this again, I was wrong. you can do this in zen the same way you would in Firefox

it's a little strange, since zen has containers set up already and it looks like they use the [firefox multi-container addon](https://addons.mozilla.org/en-US/firefox/addon/multi-account-containers/). you can adjust containers in zen's settings, but not with the full control you typically get from the addon

if you install the extension though, you can still use it normally. looks like any changes you make in the extension automatically syncs with the zen settings and vice versa

if you install the extension, go to a website, and open the extension, there should be an option to "always open this site in..." and you can choose the container
